app/Http/Controllers/Maps/DeviceDependencyController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * DependencyController.php
+ *
+ * Controller for graphing Relationships
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Thomas Berberich
+ * @author     Thomas Berberich <sourcehhdoctor@gmail.com>
+ */
+
+namespace App\Http\Controllers\Maps;
+
+use App\Models\Device;
+use Illuminate\Http\Request;
+use LibreNMS\Util\Url;
+
+class DeviceDependencyController extends MapController
+{
+    // Device Dependency Map
+    public function dependencyMap(Request $request)
+    {
+        $devices = Device::hasAccess($request->user())->with('parents')->get();
```

Comment:
More eager caching
```suggestion
        $devices = Device::hasAccess($request->user())->with(['parents', 'location'])->get();
```

Reply:
@Jellyfrog did i miss something?
i found in code nowhere usage of "location"

Reply:
changed